### PR TITLE
bugfix/update-cra-manifest-parsing

### PR DESCRIPTION
**Pull request by MasterKale**
The goal of this PR is to update django-cra-helper to support more modern versions of react-scripts. As these scripts evolved, **asset-manifest.json** files changed in shape and now offer more opportunities to more intelligently load in generated JS and CSS files within Django templates.

An effort was made to ensure API compatibility and retain support for using this library with versions as early as `react-scripts@2.1.8` (the latest, earliest known version of this library that produced the only supported syntax for **asset-manifest.json**).

However, there is a minor breaking API change to liveserver support that was required to support earlier versions while also enabling support for newer versions up through the latest `react-scripts@3.2.0`:

Before:
```html
{% if bundle_js %}
<script type="text/javascript" src="{{ bundle_js }}"></script>
{% endif %}
```

After:
```html
{% if bundle_js %}
  {% for file in bundle_js %}
<script type="text/javascript" src="{{ file }}"></script>
  {% endfor %}
{% endif %}
```

> tl;dr: `bundle_js` is now an **array** due to the introduction of code-splitting in `npm run build` artifacts starting with `react-scripts@3.0.0`. This change seems to have worked its way into older versions of `react-scripts` as well, despite my best efforts to initialize Create-React-App projects using older versions.

The good news is that, with `react-scripts@^3.2.0`, a new `entrypoints` property in **asset-manifest.json** enables easier inclusion of bundled CSS and JS assets. The README has been updated with instructions on how to leverage `entrypoints` for CRA projects using the latest `react-scripts`. This should help future-proof the library for now.

## Additional features

**Hot-reloading is now supported,** so Django views will automatically reload whenever code changes in the Create-React-App side of things triggers a reload! See the new "2. urls.py" section of the README for more info.

### CODE REVIEW

**Comment by MasterKale** [comment]
This PR should address issues #3 and #4 when it gets merged in and I deploy to a new build to PyPi.

**Comment by 10686142** [comment]
I tested this version and I finally got it working. The only thing that (maybe, I am not sure) wasn't suppose to happen, was that the terminal logs stated that the react server wasn't running while it was

**Comment by MasterKale** [comment]
Hmm, there's not a whole lot of complexity around how Django tries to determine if the CRA liveserver is running so I'm curious why, if CRA was started before Django, the helper would not thing CRA is running.

Can you help provide some more info for potentially recreating the issue?

- Can you confirm your `CRA_URL` value when starting Django?
- Are you running Create-React-App on its default port?
- The Django project has to be started _after_ `npm start` - could it be something as simple as that?

**Comment by MasterKale** [comment]
Here's some more good news: this PR also (mostly) cracks the nut that was hot-reloading Django when the Create-React-App liveserver triggered a reload.

I say "mostly" because I observed some times when the liveserver would reset the connection for some unknown reason. Manually reloading the browser window re-establishes hot-reload support till the next time a connection reset happens.

This isn't something I'm going to let block the release of this update, though.
